### PR TITLE
get Pyro version automatically

### PR DESCRIPTION
**Pull request by jpchen**
partially addresses #547, #638

this makes the version in the docs consistent with the version deployed. it is the first step in having togglable versions (those changes would be on a different branch; see discussion in issue)

### CODE REVIEW

**Comment by jpchen** [comment]
killed tests except lint and docs

File: docs/source/conf.py [hunk 1]
```diff
@@ -66,9 +68,9 @@
 #
 # The short X.Y version.
 # version = pkg_resources.require('pyro-ppl')[0].version
-version = u'0.1.2'
+version = __version__
```
Comment (fritzo) [inline docs/source/conf.py]: It seems like we'd want to be git aware here, something like
```py
if git_head_is_tag():
    version = git_tag()  # for major versions
else:
    version = '{}+{}'.format(__version__, git_commit()[:8])  # for dev
```
WDYT?

File: tutorial/source/conf.py [hunk 2]
```diff
@@ -68,9 +71,10 @@
 # built documents.
 #
 # The short X.Y version.
-version = u''
-# The full version, including alpha/beta/rc tags.
-release = u''
+# get the latest git tag, otherwise get the latest commit
+version = re.sub('^v', '', os.popen('git describe --tags').read().strip())
```
Comment (fritzo) [inline tutorial/source/conf.py]: I usually use `check_call('git describe --tags'.split()).strip()` from `subprocess` module. Is there any difference between this and `os.popen()`.

Comment (fritzo) [reply tutorial/source/conf.py]: Sorry, could you add the comment with expected output below this line?

Comment (jpchen) [reply tutorial/source/conf.py]: > When I run it I get pytorch-0.3-28-g4c52460 and I'm not sure whether this is expected

it is. basically the last tag and the commit in which it corresponds to (for those bleeding-edge users)

> Is there any difference between this and os.popen()

not anything significant.. i can change it if you have a reason that incantation is better

>  could you add the comment

done

**Comment by fritzo** [comment]
Why are tests failing?

**Comment by fritzo** [comment]
If you just let tests run to completion, then reviewers won't need to navigate to travis-ci.org to figure out why tests are failing.

**Comment by jpchen** [comment]
>  If you just let tests run to completion, then reviewers won't need to navigate to travis-ci.org to figure out why tests are failing.

sorry will do. back in my day we were very short on cores so being environmentally conscious was lauded not berated

**Comment by fritzo** [comment]
Fix lint failures.

**Pull request merged**